### PR TITLE
Moved OpenCL code in .opencl files.

### PR DESCRIPTION
**Pull request by frpays**
* created:

clsource/config.opencl
clsource/convolve1.opencl
clsource/convolve3.opencl

* made OpenCL.cc invulnerable to future clang-format

* clang-format-ted OpenCL.cc

The critical change enabling OpenCL.cc to be invulnerable to clang-format is to be to write:

`const std::string sourceCode_sgemm =
    sourceCode_blast_level3_common + sourceCode_blast_level3_xgemm_part1 +
    sourceCode_blast_level3_xgemm_part2 + sourceCode_blast_level3_xgemm_part3 +
    sourceCode_blast_level3_xgemm_batched;`

Instead of:

`const std::string sourceCode_sgemm = R"
#include ...
#include ...
`

### CODE REVIEW

File: src/neural/CL/OpenCL.cc [hunk 1]
```diff
@@ -1,527 +1,183 @@
...
-  
+
   auto weightSize = size * sizeof(decltype(converted_weights)::value_type);
   m_layers.back().weights.emplace_back(
-                                       m_opencl.m_context,
-                                       CL_MEM_COPY_HOST_PTR | CL_MEM_READ_ONLY,
-                                       weightSize,
-                                       const_cast<net_t*>(converted_weights.data()));
+      m_opencl.m_context, CL_MEM_COPY_HOST_PTR | CL_MEM_READ_ONLY, weightSize,
+      const_cast<net_t*>(converted_weights.data()));
```
Comment (dubslow) [inline src/neural/CL/OpenCL.cc]: In reference to my previous comment on different PR, if it *has* to be on more than one line, than only one argument per line is preferable IMO.

Comment (frpays) [reply src/neural/CL/OpenCL.cc]: It's clang-format google.

File: src/neural/CL/OpenCL.cc [hunk 2]
```diff
@@ -530,123 +186,87 @@ void OpenCL_Network::forward(const std::vector<net_t>& input,
...
-                inBuffer,
-                VBuffer,
-                MBuffer,
-                conv_weights,
-                nullptr,
-                bn_weights,
-                skip_in_trans, skip_next_in_trans, true);
+      convolve3(layer.channels, layer.outputs, inBuffer, inBuffer, VBuffer,
+                MBuffer, conv_weights, nullptr, bn_weights, skip_in_trans,
+                skip_next_in_trans, true);
```
Comment (dubslow) [inline src/neural/CL/OpenCL.cc]: Undecided about which is better here...

Comment (frpays) [reply src/neural/CL/OpenCL.cc]: again, It's clang-format google.

Comment (frpays) [reply src/neural/CL/OpenCL.cc]: I did not indent/reindent this file by hand.

File: src/neural/CL/OpenCL.cc (hunk 1 above)
Comment (frpays) [reply src/neural/CL/OpenCL.cc]: The point of this PR is to pass clang-format on the OpenCL.cc file.

File: src/neural/CL/OpenCL.cc [hunk 3]
```diff
@@ -53,19 +53,31 @@ const std::string sourceCode_convolve3 =
 #include "clsource/convolve3.opencl"
     ;
 
-// Important: keep the follow includes in the same orders.
-// Beware of using a formatting tool like clang-format
-// that could mess with this order.
-// The correct order is: common/part1/part2/part3/batched.
```
Comment (dubslow) [inline src/neural/CL/OpenCL.cc]: This comment is incredibly useful to anyone reading this code for the first time. Perhaps remove the comment about linters, but the part about preserving the correct order is very important and should be kept IMO.

Comment (frpays) [reply src/neural/CL/OpenCL.cc]: I removed it because I found a way around and it was no longer necessary.

Comment (frpays) [reply src/neural/CL/OpenCL.cc]: See the "critical change" in the initial comment of the PR.

Comment (frpays) [reply src/neural/CL/OpenCL.cc]: Now you can clang-format the OpenCL.cc file all you want.

Comment (frpays) [reply src/neural/CL/OpenCL.cc]: Fair enough.

Comment (mooskagh) [reply src/neural/CL/OpenCL.cc]: You can do
```
// clang-format off
around the code you don't want to be touched by clang-format
// clang-format on
```

**Pull request merged**